### PR TITLE
Devices Refactor

### PR DESCRIPTION
**Pull request by StephenHodgson**
Part 2 of the PRs

Fixed some issues with serialization of profile data. (We switched some things to use interfaces which broke some things)

All tests pass and no crashes when running in the editor.

### CODE REVIEW

File: Assets/MixedRealityToolkit/_Core/Definitions/Devices/MixedRealityInteractionMapping.cs [hunk 1]
```diff
@@ -13,59 +15,64 @@ namespace Microsoft.MixedReality.Toolkit.Internal.Definitions.Devices
     /// <remarks>One definition should exist for each physical device input, such as buttons, triggers, joysticks, dpads, and more.</remarks>
     /// </summary>
     [Serializable]
-    public struct InteractionMapping
+    public struct MixedRealityInteractionMapping : IMixedRealityInteractionMapping, ISerializationCallbackReceiver
```
Comment (SimonDarksideJ) [inline Assets/MixedRealityToolkit/_Core/Definitions/Devices/MixedRealityInteractionMapping.cs]: Please review with the changes in the other PR, shouldn't be needed as it should not be using an interface

File: Assets/MixedRealityToolkit/_Core/Definitions/MixedRealityConfigurationProfile.cs [hunk 2]
```diff
@@ -220,29 +220,5 @@ void ISerializationCallbackReceiver.OnAfterDeserialize()
         }
 
         #endregion  ISerializationCallbackReceiver Implementation
-
-        #region Mixed Reality Controller Mapping helpers
-
-        // TODO - needs validation to ensure duplicates are not added?
-
-        public MixedRealityControllerMapping GetControllerMapping(Type controllerType, Handedness hand)
```
Comment (SimonDarksideJ) [inline Assets/MixedRealityToolkit/_Core/Definitions/MixedRealityConfigurationProfile.cs]: How is this fixed by removing it>?

Comment (StephenHodgson) [reply Assets/MixedRealityToolkit/_Core/Definitions/MixedRealityConfigurationProfile.cs]: It's moved into the implemented controllers class

**Comment by StephenHodgson** [comment]
Closing this.

**Pull request closed without merging**